tests/v2/validation/configmaps/configmaps_test.go

```diff
@@ -0,0 +1,107 @@
+package configmaps
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+  "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+  "github.com/rancher/rancher/tests/framework/extensions/configmaps"
+)
+
+type ConfigmapTestSuite struct {
+  suite.Suite
+  client            *rancher.Client
+  session           *session.Session
+  cluster           *management.Cluster
+}
+
+func (cm *ConfigmapTestSuite) TearDownSuite() {
+  cm.session.Cleanup()
+}
+
+func (cm *ConfigmapTestSuite) SetupSuite() {
+  cm.session = session.NewSession()
+
+  client, err := rancher.NewClient("", cm.session)
+  require.NoError(cm.T(), err)
+  cm.client = client
+
+  clusterName := client.RancherConfig.ClusterName
+  require.NotEmptyf(cm.T(), clusterName, "Cluster name to install should be set")
+  clusterID, err := clusters.GetClusterIDByName(cm.client, clusterName)
+  require.NoError(cm.T(), err, "Error getting cluster ID")
+  cm.cluster, err = cm.client.Management.Cluster.ByID(clusterID)
+  require.NoError(cm.T(), err)
+}
+
+func (cm *ConfigmapTestSuite) TestConfigMapCreate() {
+  subSession := cm.session.NewSession()
+  defer subSession.Cleanup()
+
+  configMapClient := cm.client.Steve.SteveType(configmaps.ConfigMapSteveType)
+  
+  log.Info("Creating a configmap")
+  configmapName := namegenerator.AppendRandomString(cmName)
+  configmapObj, err := createConfigmap(*configMapClient, configmapName, namespace, nil, labels, data)
+  require.NoError(cm.T(), err)
+
+  log.Info("Validating configmap was created")
+  assert.Contains(cm.T(), configmapObj.Name, configmapName)
+}
```

Comment:
Once this is moved to configmaps.go, we can then add a validation for the annotations or labels added in this test as well. https://github.com/rancher/rancher/blob/ad54ec3510b3eeed063b7a83edea1f3357879fe8/tests/v2/validation/configmaps/configmaps_test.go#L76